cfme/tests/infrastructure/test_host.py

```diff
@@ -540,26 +541,51 @@ def test_add_ipmi_refresh(appliance, setup_provider):
 
 
 @test_requirements.infra_hosts
-@pytest.mark.parametrize("crud_action", ['edit_from_hosts', 'edit_from_details', 'remove'])
+@pytest.mark.meta(automates=[1634794])
+@pytest.mark.parametrize("crud_action", ['edit_from_hosts', 'edit_from_details', 'cancel',
+                                         'nav_away_changes', 'nav_away_no_changes', 'remove'])
 def test_infrastructure_hosts_crud(appliance, setup_provider, crud_action):
     """
     Polarion:
         assignee: prichard
         casecomponent: Infra
         caseimportance: low
         initialEstimate: 1/6h
+    Bugzilla:
+        1634794
     """
     host = appliance.collections.hosts.all()[0]
+    if crud_action in ['cancel', 'nav_away_changes', 'nav_away_no_changes']:
+        try:
+            existing_custom_id = navigate_to(host, 'Details').entities.summary(
+                "Properties").get_text_of("Custom Identifier")
+        except NameError:
+            existing_custom_id = None
     if crud_action != 'remove':
-        stamp = fauxfactory.gen_alphanumeric()
-        new_custom_id = f'Edit host data. {stamp}'
-
-        with update(host, from_details=(crud_action == 'edit_from_details')):
-            host.custom_ident = new_custom_id
-
-        assert host.custom_ident == new_custom_id
-        assert navigate_to(host, 'Details').entities.summary("Properties").get_text_of(
-            "Custom Identifier") == new_custom_id
+        new_custom_id = f'Edit host data. {fauxfactory.gen_alphanumeric()}'
+        try:
+            with update(host,
+                        from_details=(crud_action == 'edit_from_details'),
+                        cancel=(crud_action == 'cancel'),
+                        nav_away=('nav_away' in crud_action),
+                        changes=('changes' in crud_action and 'no' not in crud_action)
```

Comment:
Whether you keep these new kwargs in `update()` or move them elsewhere, it's better to be explicit about what values of `crud_action` correspond to which kwarg values:

```
'nav_away': crud_action in ('nav_away_changes', 'nav_away_no_changes'),
'changes': crud_action == 'nav_away_changes'
```

Reply:
This does make things clearer. Updated as suggested.